napi-inl.h

```diff
@@ -1432,16 +1434,49 @@ inline Error Error::New(napi_env env, const std::string& message) {
   return Error::New<Error>(env, message.c_str(), message.size(), napi_create_error);
 }
 
-inline Error::Error() : Object(), _message(nullptr) {
+inline Error::Error() : ObjectReference(), _message(nullptr) {
 }
 
-inline Error::Error(napi_env env, napi_value value) : Object(env, value) {
+inline Error::Error(napi_env env, napi_value value) : ObjectReference(env, nullptr) {
+  if (value != nullptr) {
+    napi_status status = napi_create_reference(env, value, 1, &_ref);
+
+    // Avoid infinite recursion in the failure case.
+    // Don't try to construct & throw another Error instance.
+    assert(status == napi_ok);
```

Comment:
Probably. See also a few other places in this `Error` class where error-handling code can't throw errors.

We'd need a `napi_fatal_error()` API or similar.